arrow/src/json/reader.rs

```diff
@@ -942,6 +943,30 @@ impl Decoder {
         ))
     }
 
+    #[allow(clippy::unnecessary_wraps)]
+    fn build_primitive_array_using_format_string<T>(
+        &self,
+        rows: &[Value],
+        col_name: &str,
+        format_string: &str,
+    ) -> Result<ArrayRef>
+    where
+        T: ArrowNumericType + Parser,
+        T::Native: num::NumCast,
```

Comment:
Do we need this trait bound? It may be left over from `build_primitive_array` 🤔 

It seemed to compile just fine like this for me:

```suggestion
        T: Parser,
```